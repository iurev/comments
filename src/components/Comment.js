import React from 'react';
import styled from 'styled-components'

const Root = styled.div`
  margin-bottom: 10px;
`

const Title = styled.div`
  color: ${p => p.theme.darkFontColor};
  font-size: 24px;
  margin-bottom: 5px;
`

const Text = styled.div`
  color: ${p => p.theme.mediumFontColor};
  font-size: 14px;
`

const Comment = ({ title, text }) => (
  <Root>
    <Title>{title}</Title>
    <Text dangerouslySetInnerHTML={{
      __html: text
    }}>
    </Text>
  </Root>
)

export default Comment;
