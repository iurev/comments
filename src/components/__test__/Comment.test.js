import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Comment from '../Comment';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Comment />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('with all attrs', () => {
  const tree = renderer
    .create(<Comment
      title="test"
      text="test"
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
