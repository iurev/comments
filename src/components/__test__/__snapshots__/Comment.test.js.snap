// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with all attrs 1`] = `
<div
  className="sc-bdVaJa jWKkjw"
>
  <div
    className="sc-bwzfXH fIVpWC"
  >
    test
  </div>
  <div
    className="sc-htpNat kqVAcx"
    dangerouslySetInnerHTML={
      Object {
        "__html": "test",
      }
    }
  />
</div>
`;
